### DISCUSSION TITLE
Sayfada çok veri var ve tarayıcı donuyor

### DISCUSSION THREAD

**Discussion by kostok** [ademilter/homepage#42] [Q&A] [answered]
Selamlar benim iki sorum olacaktı.
1) http://en.arguman.org/c-is-better-for-programming-pc-apps-than-java 
linkteki sayfada boş bir alana tıklayıp mause basılı tutarak sayfa kaydırma olayının bir adı ve yöntemi var mıdır? Yoksa js ile mause eventlerinden yakalayıp scroll pozisyonlarını mı değiştireceğiz.

2) Verileri görüntülemek için divlerden oluşturduğum bir datagrid var json datalarını parçalayıp element oluşturup satır satır ekliyorum ancak tüm veriler ekrana geldikten sonra tarayıcı çok kasıyor scroll dahi oynamıyor sayfa donuyor. Ekranda çok veri var ve bunların hepsinin yüklü olması ve kullanıcının bunları görmesi gerekiyor. Bu problemi nasıl çözebilirim tavsiyeleriniz var mı?

Buraya yazmadan önce araştırma yaptım fakat bir sonuca varamadım cevaplarınız için şimdiden teşekkür ederim.

**Comment by ademilter** [comment, marked as answer]
selam,

**ilk soruya cevap;**
evet, event'leri dinliyerek istediğini yapabilirsin. content için bi kapsayıcı oluştur ve dört kenara tarayıcının boyutlarına oranla boşluklar ver. daha sonra sayfa yüklendiğinde content hep ortaya gelecek şekilde scroll pozisyonunu ayarla. ve sürükle bırak ile canvas'ı istediğin gibi sürükle. yapamazsan tekrar yaz örnek kod yazmaya çalışsalım

![caps](https://user-images.githubusercontent.com/187922/48678464-cce44800-eb94-11e8-894c-f0370aae4035.png)



**ikinci soruya cevap;**
bu yöntemin adı [virtual-scroll](https://github.com/valdrinkoshi/virtual-scroller).
sahnenein dışına çıkan elemanları DOM'dan silerek ciddi performans artışı sağlanıyor.
kullandığın frameworke göre eklenti kullanabililirsin. mesela ben vue için bir tane kullanmıştım
